interp/notation_ops.ml

```diff
@@ -48,6 +48,22 @@ let replace_var i j var = j :: List.remove Id.equal i var
    [t2] is finer than [t1]); when [lt] is false, tell if [t1] is the
    same pattern as [t2] *)
 
+let compare_universe_instances lt us1 us2 =
+  match us1, us2 with
+  | None, None -> not lt
+  | Some _, None -> true
+  | None, Some _ -> false
+  | Some l1, Some l2 ->
+     CList.for_all2eq (fun u1 u2 -> (* <= *)
+         match u1, u2 with
+         | UAnonymous {rigid=true}, UAnonymous {rigid=true} -> true
+         | UAnonymous {rigid=false}, UAnonymous {rigid=false} -> true
+         | UAnonymous _, UAnonymous _ -> false
+         | UNamed _, UAnonymous _ -> true
+         | UAnonymous _, UNamed _ -> false
+         | UNamed _, UNamed _ -> glob_level_eq u1 u2) l1 l2
+    && not (lt && CList.equal glob_level_eq l1 l2)
```

Comment:
```suggestion
         | UNamed _, UAnonymous _ -> not lt
         | UAnonymous _, UNamed _ -> false
         | UNamed _, UNamed _ -> glob_level_eq u1 u2) l1 l2
```
IIUC

Reply:
My reasoning was that `UNamed _` should always be strictly more specific than `UAnonymous _`.
If you move the `not lt` to the equal anonymous cases, you run into the missing `CList` combinator for comparing lists: list `a` is strictly less than list `b` if they are pointwise less than or equal and at least one element is strictly less. You want `@{a _ _} < @{_ _ _}`.
That is why I compute `<=` with `CList.for_all2eq`, and then handle the equality case separately.